rollouts/controllers/remotesync_controller.go

```diff
@@ -143,53 +155,60 @@ func (r *RemoteRootSyncReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 				// Make sure we stop any watches that are no longer needed.
 				logger.Info("Pruning watches")
-				r.pruneWatches(req.NamespacedName, &remoterootsync.Spec.ClusterRef)
+				r.pruneWatches(req.NamespacedName, &remotesync.Spec.ClusterRef)
 			}
 
 			// remove our finalizer from the list and update it.
-			controllerutil.RemoveFinalizer(&remoterootsync, myFinalizerName)
-			if err := r.Update(ctx, &remoterootsync); err != nil {
+			controllerutil.RemoveFinalizer(&remotesync, myFinalizerName)
+			if err := r.Update(ctx, &remotesync); err != nil {
 				return ctrl.Result{}, fmt.Errorf("failed to update %s after delete finalizer: %w", req.Name, err)
 			}
 		}
 		// Stop reconciliation as the item is being deleted
 		return ctrl.Result{}, nil
 	}
 
-	syncStatus, syncError := r.syncExternalSync(ctx, &remoterootsync)
+	syncStatus, syncError := r.syncExternalSync(ctx, &remotesync)
 
-	if err := r.updateStatus(ctx, &remoterootsync, syncStatus, syncError); err != nil {
+	if err := r.updateStatus(ctx, &remotesync, syncStatus, syncError); err != nil {
 		logger.Error(err, "Failed to update status")
 		return ctrl.Result{}, err
 	}
 
 	return ctrl.Result{}, syncError
 }
 
-func (r *RemoteRootSyncReconciler) syncExternalSync(ctx context.Context, rrs *gitopsv1alpha1.RemoteRootSync) (string, error) {
-	syncName := rrs.Name
-	clusterRef := &rrs.Spec.ClusterRef
+func (r *RemoteSyncReconciler) syncExternalSync(ctx context.Context, remotesync *gitopsv1alpha1.RemoteSync) (string, error) {
+	syncName := remotesync.Name
+	clusterRef := &remotesync.Spec.ClusterRef
 
 	dynCl, err := r.getDynamicClientForCluster(ctx, clusterRef)
 	if err != nil {
 		return "", fmt.Errorf("failed to create client: %w", err)
 	}
 
-	if err := r.patchRootSync(ctx, dynCl, syncName, rrs); err != nil {
+	var namespace string
+	if remotesync.Spec.Type == gitopsv1alpha1.TemplateTypeRepoSync {
+		namespace = remotesync.Namespace
+	} else {
+		namespace = rSyncNamespace
+	}
+
+	if err := r.patchSync(ctx, dynCl, syncName, remotesync, namespace); err != nil {
 		return "", fmt.Errorf("failed to create/update sync: %w", err)
 	}
 
-	r.setupWatches(ctx, rrs.Name, rrs.Namespace, rrs.Spec.ClusterRef)
+	r.setupWatches(ctx, remotesync.Name, remotesync.Namespace, remotesync.Spec.ClusterRef)
 
-	syncStatus, err := checkSyncStatus(ctx, dynCl, syncName)
+	syncStatus, err := checkSyncStatus(ctx, dynCl, syncName, remotesync.Spec.Type, namespace)
 	if err != nil {
-		return "", fmt.Errorf("faild to check status: %w", err)
+		return "", fmt.Errorf("failed to check status: %w", err)
```

Comment:
Now that namespace is critical and different objects involved, I think will be good a add a note about the relationship between the namespaces of different object involved here: Rollout.Namespace, RS.Namespace, ExternalSyncObject.Namespace.

Reply:
Added some comments explaining the relationship between these namespaces.